server/controllers/availability.js

```diff
@@ -0,0 +1,139 @@
+const PetSitter = require("../models/PetSitter")
+const Schedule = require("../models/Schedule");
+const asyncHandler = require("express-async-handler");
+const Availability = require("../models/Availability");
+
+// @route POST /availability/schedule/:scheduleId
+// @desc create or update a schedule
+// @access Private
+exports.createSchedule = asyncHandler(async (req, res, next) => {
+  const petsitter = await PetSitter.findOne(req.user);
+  const availability = await Availability.findById(petsitter.availabilityId);
+  const scheduleIds = await availability.scheduleIds
+  const { name, MondayFrom, MondayTo, TuesdayFrom, TuesdayTo, WednesdayFrom, WednesdayTo, ThursdayFrom, ThursdayTo, FridayFrom, FridayTo, SaturdayFrom, SaturdayTo, SundayFrom, SundayTo } = req.body;
+  
+  const newPeriods = {
+    monday: {
+      startTime: MondayFrom,
+      endTime: MondayTo,
+    },
+    tuesday: {
+      startTime: TuesdayFrom,
+      endTime: TuesdayTo,
+    },
+    wednesday: {
+      startTime: WednesdayFrom,
+      endTime: WednesdayTo,
+    },
+    thursday: {
+      startTime: ThursdayFrom,
+      endTime: ThursdayTo,
+    },
+    friday: {
+      startTime: FridayFrom,
+      endTime: FridayTo,
+    },
+    saturday: {
+      startTime: SaturdayFrom,
+      endTime: SaturdayTo,
+    },
+    sunday: {
+      startTime: SundayTo,
+      endTime: SundayFrom,
+    },
+  };
+
+  if (scheduleIds.includes(req.params.scheduleId)) {
+    schedule = await Schedule.findById(req.params.scheduleId)
+    schedule.name = name;
+    schedule.periods = newPeriods;
+    await schedule.save()
+
+    res.status(201).json({
+      success: {
+        schedule
+      }
+    })
+  } else {
+      const newSchedule = await Schedule.create({
+        name,
+        periods: newPeriods
+      });
+
+      availability.scheduleIds.push(newSchedule._id)
+      await availability.save()
+  
+      res.status(201).json({
+        success: {
+          newSchedule
+        }
+      })
+  }
+});
+
+// @route GET /availability/schedule/active
+// @desc get active schedule
+// @access Private
+exports.activeSchedule = asyncHandler(async (req, res, next) => {
+  const petsitter = await PetSitter.findOne(req.user);
+  const availability = await Availability.findById(petsitter.availabilityId);
+  const activeScheduleId = await availability.activeScheduleId;
+  const activeSchedule = await Schedule.findById(activeScheduleId)
```

Comment:
In this case, do we care much about the availability or do we care more just about the schedule itself - kind of on me for the weird structure but we don't need to make a query to availability. We can query the schedule itself and from there check if it doesn't exist then reject. 

We're making 2 queries when we can really only make one 👍🏻 🙂

Reply:
Applicable throughout this whole file 😄